### ISSUE TITLE
Quests become replayable upon server restart

### ISSUE THREAD

**Issue by mrconno** [PikaMug/Quests#1471]
**What is your Quests version / build number (do _not_ say "latest")?**

Quests 4.0.0-rc.2

**Describe the problem and how to replicate it. Any console errors to post _using Pastebin.com_?**

When a player completes a quest and the server restarts, it says they can do the quest again, however they keep the quest point they got from completing it the first time. I tried what you suggested in my earlier post #1469 which was too have the player log out and in before restarting the server. The quests were still reset, as if they had never done them (besides still having the rewards/ quest points)
I just want to make it so certain quests can not be replayed.
I'd like to add that the server restarts on its own daily so the issue is all the players would get to redo all the quests everyday ruining the economy of the server and the quest progressions via quest points.
Thanks!

Player does quest "mine diamonds" (mine 5 diamonds, quest completes, 1 quest point awarded) 
Player uses /quests stats. it states, quest points: 1, quests completed: mine diamonds
Server restarts. Player logs in and types /quests stats, quest points: 1, quests completed: none
Player can now replay the quest and get more quest points and rewards.

TLDR: Quests uncomplete themselves upon server restart and players can redo them for more quest points and rewards. 

**If applicable, can you provide an example quest from quests.yml _using Pastebin.com_?**

```yaml
custom4:
    name: Mine Diamonds
    ask-message: Hi adventurer! This quest is very simple. Please mine 5 diamonds.
      Happy mining!
    finish-message: Great job mining! Enjoy your reward.
    npc-giver-id: 28
    stages:
      ordered:
        '1':
          break-block-names:
          - Diamond_ore
          break-block-amounts:
          - 5
          break-block-durability:
          - 0
    rewards:
      items:
      - ==: org.bukkit.inventory.ItemStack
        v: 2580
        type: DIAMOND_PICKAXE
        meta:
          ==: ItemMeta
          meta-type: UNSPECIFIC
          display-name: Grace's Special Pickaxe
          enchants:
            MENDING: 1
      money: 10000
      quest-points: 1
      exp: 150
    planner:
      redo-delay: false
```

**Comment by mrconno** [comment]
Sorry heres the bottom part on pastbin
https://pastebin.com/fbsJcjBs

**Comment by PikaMug** [comment]
@mrconno Thanks for the report. Please **backup your _/Quests/data_** folder and try the following development build:
https://ci.codemc.io/job/PikaMug/job/Quests/166/

As there have been many recent changes to player data operations, the build may resolve this issue. If you find this is not the case, please reply below with any relevant information.
